### ISSUE TITLE
Ringtone won't stop playing on open conversation

### ISSUE THREAD

**Issue by Quintasan** [signalapp/Signal-Android#7129]
I have:
- [X] searched open and closed issues for duplicates
- [X] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
If I receive a message in a conversation I have already open the ringtone goes off and won't shut up untill I kill Signal. I can record a movie in case it's not obvious what happens.

### Steps to reproduce
- Open Signal
- Open a conversation
- Ask that person to send you a message
- The ringtone won't shut up even if you reopen the conversation

**Actual result:** Ringtone goes off and stops only after it's played in full length.
**Expected result:** I'd expect the ringtone to stop after I tap anywhere on the screen or after I go back and reopen the conversation.

### Device info
**Device:** Xiaomi Mi 5
**Android version:** 7.1.2
**Signal version:** 4.11.5

### Link to debug log
https://gist.github.com/707f3ee517560d89cab64dd027f83077

**Comment by raffthesweetling13** [comment]
I was able to reproduce this on an Android Studio emulator. 

I got a pretty shoddy fix working, and wanted to get some feedback before I continued working.

My solution was to have the activities that call [setVisibleThread](https://github.com/WhisperSystems/Signal-Android/blob/master/src/org/thoughtcrime/securesms/notifications/MessageNotifier.java#L96) also pass in a view that could be tapped that would stop the ringtone. Then in `MessageNotifier` I can add an `OnTouchListener` that will call [Ringtone.stop](https://developer.android.com/reference/android/media/Ringtone.html#stop()) if the view that was passed in is touched. 

Along with some general feedback/recommendations about that approach, some other questions I have:
1. What activities should this be enabled on? Currently only the `ConversationActivity` and the `MessageDetailsActivity` use the `setVisibleThread` method mentioned earlier. 
2. Where on the screen should tapping cause the ringtone to get silenced? 
3. Any other actions that should stop the ringtone? 

- Pressing the home button while in a view that contains the new message? 
- What about when entering a view that contains the new message? 
- So far I'm just using an `OnTouchListener` what about clicks? (Android Studio gives me a warning about implementing `onClick` whenever you implement `onTouch` for accessibility reasons.)

4. Any thoughts on storing a static reference to the view that the listener is set on in `MessageNotifier`?
